storage.go

```diff
@@ -86,13 +92,41 @@ func (s *Storage) createAppend(ctx context.Context, path string, opt pairStorage
 	return o, nil
 }
 
+func (s *Storage) createDir(ctx context.Context, path string, opt pairStorageCreateDir) (o *Object, err error) {
+	rp := s.getAbsPath(path)
+
+	res, err := s.client.CreateFolderV2(&files.CreateFolderArg{
+		Path: rp,
+	})
+	if err != nil && checkError(err, files.CreateFolderErrorPath, files.WriteErrorConflict, files.WriteConflictErrorFolder) {
+		// Omit `path/conflict/folder` (`dir exists` related) error here.
+		err = nil
+	}
+	if err != nil {
+		return
+	}
+
+	o = s.newObject(true)
+	o.Mode = ModeDir
+	if res != nil {
+		o.ID = res.Metadata.Id
+		o.Path = res.Metadata.Name
+	} else {
+		o.ID = rp
+		o.Path = path
+	}
+
+	return o, nil
+}
+
 func (s *Storage) delete(ctx context.Context, path string, opt pairStorageDelete) (err error) {
 	rp := s.getAbsPath(path)
 
 	input := &files.DeleteArg{
 		Path: rp,
 	}
 
+	// If the path is a folder, all its contents will be deleted too.
```

Comment:
But `Delete` requires only remove one object.

Reply:
Yes. Seems there's no API or configuration to refuse deleting an unempty dir. Maybe we can `ListFolder` before to check whether the folder is empty?